netdcli/netdcli.go

```diff
@@ -498,26 +508,53 @@ func executeOpts(opts *cliOpts) error {
 	case cliOperGet:
 		err = coreState.Read(opts.idStr)
 		if err != nil {
-			log.Fatalf("Failed to read %s. Error: %s", opts.construct.Get(), err)
-		} else {
-			content, err := json.MarshalIndent(coreState, "", "  ")
-			if err != nil {
-				log.Fatalf("Failed to marshal corestate %+v", coreState)
-			}
-			fmt.Println(string(content))
+			log.Errorf("Failed to read %s. Error: %s", opts.construct.Get(), err)
+			return err
 		}
+		content, err := json.MarshalIndent(coreState, "", "  ")
```

Comment:
I agree. However, right now we are just using it for netdcli to print the info on screen. And since we have now moved away from screen-scraping with #98 we don't have a need yet to make a common function for this.

Actually I realized, I should have been more clear about some minor changes in this diff. This change is actually related to code refactor to make go-lint happy after some of my changes to return error to the calling function instead of calling Fatalf() .